mmv1/products/bigqueryconnection/Connection.yaml

```diff
@@ -299,18 +299,48 @@ properties:
       - !ruby/object:Api::Type::String
         name: 'database'
         description:
-          Cloud Spanner database in the form `project/instance/database'
+          Cloud Spanner database in the form `project/instance/database'.
         required: true
       - !ruby/object:Api::Type::Boolean
         name: 'useParallelism'
         description:
-          If parallelism should be used when reading from Cloud Spanner
+          If parallelism should be used when reading from Cloud Spanner.
+      - !ruby/object:Api::Type::Integer
+        name: 'maxParallelism'
+        description:
+          Allows setting max parallelism per query when executing on Spanner independent compute
+          resources. If unspecified, default values of parallelism are chosen that are dependent on
+          the Cloud Spanner instance configuration. `useParallelism` and `useDataBoost` must be set
+          when setting max parallelism.
+        required_with:
+          - useDataBoost
+          - useParallelism
```

Comment:
This need to be the full path to the parameter.

see
https://github.com/GoogleCloudPlatform/magic-modules/blob/b7e0a8c9627221adef70ae7c6693821e55a26211/mmv1/products/secretmanager/Secret.yaml#L223

Reply:
Thanks for the catch, updated and I can see the generated `.go` file having e.g. `RequiredWith: []string{cloud_spanner.0.use_parallelism}`, but there is no TF validation error when I run `apply` and the request goes to the server. Is that expected?

Reply:
I was able to get a plan time error

```hcl
resource "google_bigquery_connection" "meep" {
  provider= google-local
   connection_id = "<%= ctx[:vars]['connection_id'] %>"
   location      = "US"
   friendly_name = "👋"
   description   = "a riveting description"
   cloud_spanner {
      database        = "<%= ctx[:vars]['database'] %>"
      use_data_boost  = true
      max_parallelism = 100
   }
}
```

```
scottsuarez-macbookpro2:10-18 scottsuarez$ terraform plan 
╷
│ Error: Missing required argument
│ 
│   with google_bigquery_connection.meep,
│   on main.tf line 9, in resource "google_bigquery_connection" "meep":
│    9:       use_data_boost  = true
│ 
│ "cloud_spanner.0.use_data_boost": all of `cloud_spanner.0.use_data_boost,cloud_spanner.0.use_parallelism` must be specified
╵
╷
│ Error: Missing required argument
│ 
│   with google_bigquery_connection.meep,
│   on main.tf line 10, in resource "google_bigquery_connection" "meep":
│   10:       max_parallelism = 100
│ 
│ "cloud_spanner.0.max_parallelism": all of
│ `cloud_spanner.0.max_parallelism,cloud_spanner.0.use_data_boost,cloud_spanner.0.use_parallelism` must be specified
╵
````

Reply:
Note that required with is a one way dependency.

Reply:
Acknowledged. That's what we want.